### ISSUE TITLE
Tooling: consider switching to jsonc-eslint-parser

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [JoshuaKGoldberg/eslint-plugin-package-json#43]
@ota-meshi's https://github.com/ota-meshi/jsonc-eslint-parser looks _excellent_: well-documented, well-tested, and well-used by https://www.npmjs.com/package/eslint-plugin-jsonc.

I imagine this might unblock #40? At the very least, I like the idea of using the standard community method for working with JSON.

**Comment by ota-meshi** [comment]
Hi, @JoshuaKGoldberg!
This may be off-topic, but please let me comment.
I have a plugin that only checks dependencies in package.json.
https://github.com/ota-meshi/eslint-plugin-node-dependencies
When using jsonc-eslint-parser, that plugin's rules can be used together.
However, in that case those rules check package.json, so I would like the rules for `eslint-plugin-node-dependencies` to be merged into `eslint-plugin-package-json` if possible.
What do you think?

**Comment by JoshuaKGoldberg** [comment]
Yes, definitely - thanks for surfacing! That would be great. I hadn't seen `eslint-plugin-node-dependencies` but it has quite a few rules that look very good to have. Added it to the table in #42.

I'll take this as general confirmation that `jsonc-eslint-parser` is a good step forward for this package 🙂. Will try to get a draft PR up soon.
